lib/entities/quote/DutchV2Quote.ts

```diff
@@ -0,0 +1,221 @@
+    const nonce = this.nonce ?? generateRandomNonce();
+
+    const builder = orderBuilder
+      .deadline(deadline)
+      .swapper(ethers.utils.getAddress(this.request.config.swapper))
+      .nonce(BigNumber.from(nonce))
+      .cosigner(LABS_COSIGNER)
+      // empty cosignature so we can serialize the order
+      .cosignature(ethers.constants.HashZero)
+      .input({
+        token: this.tokenIn,
+        startAmount: this.amountInStart,
+        endAmount: this.amountInEnd,
+      });
+
+    const outputs = getPortionAdjustedOutputs(
+      {
+        token: this.tokenOut,
+        startAmount: this.amountOutStart,
+        endAmount: this.amountOutEnd,
+        recipient: this.request.config.swapper,
+      },
+      this.request.info.type,
+      this.request.info.sendPortionEnabled,
+      this.portion
+    );
+    outputs.forEach((output) => builder.output(output));
+
+    return builder.buildPartial();
+  }
+
+  public toLog(): LogJSON {
+    return {
+      tokenInChainId: this.chainId,
+      tokenOutChainId: this.chainId,
+      requestId: this.requestId,
+      quoteId: this.quoteId,
+      tokenIn: this.tokenIn,
+      tokenOut: this.tokenOut,
+      amountIn: this.amountInStart.toString(),
+      amountOut: this.amountOutStart.toString(),
+      endAmountIn: this.amountInEnd.toString(),
+      endAmountOut: this.amountOutEnd.toString(),
+      amountInGasAdjusted: this.amountInStart.toString(),
+      amountInGasAndPortionAdjusted:
+        this.request.info.type === TradeType.EXACT_OUTPUT ? this.amountInGasAndPortionAdjusted.toString() : undefined,
+      amountOutGasAdjusted: this.amountOutStart.toString(),
+      amountOutGasAndPortionAdjusted:
+        this.request.info.type === TradeType.EXACT_INPUT ? this.amountOutGasAndPortionAdjusted.toString() : undefined,
+      swapper: this.swapper,
+      routing: RoutingType[this.routingType],
+      slippage: parseFloat(this.request.info.slippageTolerance),
+      createdAt: timestampInMstoSeconds(parseInt(this.createdAtMs)),
+      createdAtMs: this.createdAtMs,
+      portionBips: this.portion?.bips,
+      portionRecipient: this.portion?.recipient,
+      portionAmountOutStart: this.portionAmountOutStart.toString(),
+      portionAmountOutEnd: this.portionAmountOutEnd.toString(),
+    };
+  }
+
+  getPermitData(): PermitTransferFromData {
+    return this.toOrder().permitData();
+  }
+
+  public get amountOut(): BigNumber {
+    return this.amountOutStart;
+  }
+
+  public get amountIn(): BigNumber {
+    return this.amountInStart;
+  }
+
+  // The number of seconds from endTime that the order should expire
+  public get deadlineBufferSecs(): number {
+    if (this.request.config.deadlineBufferSecs !== undefined) {
+      return this.request.config.deadlineBufferSecs;
+    }
+
+    switch (this.chainId) {
+      case 1:
+        // 10 blocks from now
+        // to cover time to sign, run secondary auction, and some blocks for decay
+        return 120;
```

Comment:
Maybe, but don't think its too important for start - vast majority of trades should fill in exclusivity. Kind of want to see how it performs before doing too much fancy stuff to parameterization. Leaning liberal to start for the public decay, to avoid failures in that small % of fades

Reply:
this will likely be a research topic